### ISSUE TITLE
Cluster comparison score

### ISSUE THREAD

**Issue by snayfach** [antismash/antismash#365]
Thanks for the great tool and paper! I've cloned antismash 6 from github and run it with the --cc-mibig flag on several metagenomic assemblies. I have a few questions on the new cluster comparison module.

I may have missed it, but I didn't see any detailed description of how the similarity scores are computed. Is there documentation on that? My understanding is that the algorithm uses a combination of sequence similarity, sharing of biosynthetic modules, and synteny. Does the algorithm also take into account the number of shared genes/domains? I'm worried that a high score could be obtained from sharing just a single biosynthetic gene.

My concern is related to observations I've made. In one example (see attachment), the top ranked reference BGC had a similarity score of 0.54 with only a single shared gene. Further down the ranked list, there was another BGC with 14 shared genes but a similarity score of only 0.50. The latter BGC is clearly more similar to my query than the top ranked BGC with a higher score. My results can be downloaded here: https://www.dropbox.com/s/4nx59wlt1mbyeo2/ERR720048.zip

Also, is there a recommended similarity cutoff for predicting that two BGCs have the same function?

Thanks again!

**Comment by SJShaw** [comment]
A hard similarity cutoff for two regions having the same function doesn't really exist. There's just so many variables at play with cluster type, hybrids, which substrates are loaded (if substrates are loaded), and so on. Even with a perfect score for the 'Reference in Query' mode that the MIBiG comparison uses, you might still have extra functionality in your query that makes the two areas functionally different. On the other hand, a non-perfect score can result from the reference cluster containing unrelated genes, but would be a perfect match if the reference was correctly trimmed.

As a rough guide, I'd say anything less than 0.7 is in "vaguely related" territory and anything above 0.9 is looking really good. Of course this is still the first step, you'll still have to look at those higher scoring references to see if it makes sense with your particular input and whatever cluster type is involved. 

There will be a paper covering the details of the calculations at some stage. There are a few edge cases that can crop up and we're still tweaking as the appear. Single gene hits are one of those, but as there's some actual single gene entries in MIBiG, the options for handling them are more limited.
As a brief summary, only genes with 25% identity and 30% coverage are linked and only these genes are considered for the sequence similarity score. The ordering metric is based on the count, lengths, and strands of matching segments ("blocks" of preserved synteny).
For the components metric, the full cluster is used, so that's biosynthetic domains, NRPS/PKS modules, gene functions, and so on.  This makes it extremely unlikely to have a high score with a single gene match (barring those cases where the entire reference cluster is only a single gene, of course).

For your input, the lower scoring cluster has more genes that are similar, but the identities of those matches that aren't really any better than the top hit. The components score quite similarly between the two, as your query region is missing a number of biosynthetic-additional genes and their contents of the reference arylpolyene. The ordering is penalised by the gaps in the arylpolyene hits, and ignored in the top hit, but even removing the ordering metric completely to treat them similarly, the final scores would really be very similar (0.56 vs 0.54). 

So to sum up, the ordering here could be better. There are a couple of minor tweaks we'll be doing, which may bring up the other hit to be the top hit, but it's highly likely that the score for it still won't look great and would still need a manual check to really see how close the products would be.

**Comment by snayfach** [comment]
I really appreciate the detailed reply and look forward to reading your cluster comparison tool paper!  My 2c is that the final score should incorporate the total coverage of the query and/or reference BGC so that a very high score can't be obtained from just a few genes.
